mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -155,201 +245,483 @@ public void testSimpleAttachment2() throws MessagingException, IOException {
+
+        assertThat(name.startsWith("e_Pubblicita_e_vietata_Milano9052")).isTrue();
+        
+        assertThat(new File(folderPath + name)).hasContent(expectedAttachmentContent);
+    }
+
+    @Test
+    public void serviceShouldSaveFilenameAttachmentAndFileContentInCustomAttribute() throws MessagingException, IOException {
+        StripAttachment mailet = new StripAttachment();
 
-        Assert.assertEquals(1, c.size());
+        String customAttribute = "my.custom.attribute";
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("remove", "matched")
+                .setProperty("directory", folderPath)
+                .setProperty("pattern", ".*\\.tmp")
+                .setProperty("attribute", customAttribute)
+                .build();
+        mailet.init(mci);
+        
+        MimeMessage message = new MimeMessage(Session
+                .getDefaultInstance(new Properties()));
 
-        String name = c.iterator().next();
-        // System.out.println("--------------------------\n\n\n");
-        // System.out.println(name);
+        MimeMultipart multiPart = new MimeMultipart();
+        MimeBodyPart part = new MimeBodyPart();
+        part.setText("simple text");
+        multiPart.addBodyPart(part);
+        String expectedKey = "10.tmp";
+        multiPart.addBodyPart(createAttachmentBodyPart("\u0023\u00A4\u00E3\u00E0\u00E9", expectedKey));
+        multiPart.addBodyPart(createAttachmentBodyPart("\u0014\u00A3\u00E1\u00E2\u00E4", "temp.zip"));
+        
+        message.setSubject("test");
+        message.setContent(multiPart);
+        message.saveChanges();
+
+        Mail mail = FakeMail.builder()
+                .mimeMessage(message)
+                .build();
 
-        Assert.assertTrue(name.startsWith("e_Pubblicita_e_vietata_Milano9052"));
+        mailet.service(mail);
 
-        File f = new File("./" + name);
-        try {
-            InputStream is = new FileInputStream(f);
-            String savedFile = toString(is);
-            is.close();
-            Assert.assertEquals(body, savedFile);
-        } finally {
-            FileUtils.deleteQuietly(f);
-        }
+        @SuppressWarnings("unchecked")
+        Map<String, byte[]> saved = (Map<String, byte[]>) mail.getAttribute(customAttribute);
+        assertThat(saved).hasSize(1);
+        assertThat(saved).containsKey(expectedKey);
     }
 
     @Test
-    public void testToAndFromAttributes() throws MessagingException,
-            IOException {
-        Mailet strip = new StripAttachment();
+    public void initShouldThrowWhenPatternAndNotPatternAreNull() throws MessagingException {
+        Mailet mailet = new StripAttachment();
+
         FakeMailetConfig mci = FakeMailetConfig.builder()
                 .mailetName("Test")
-                .setProperty("attribute", "my.attribute")
+                .setProperty("directory", folderPath)
```

Comment:
You can even remove the lines : 
 - .setProperty("directory", folderPath)
 -  .setProperty("remove", "all")